airflow/contrib/operators/emr_add_steps_operator.py

```diff
@@ -68,12 +68,14 @@ def __init__(
         self.steps = steps
 
     def execute(self, context):
-        emr = EmrHook(aws_conn_id=self.aws_conn_id).get_conn()
+        emr_hook = EmrHook(aws_conn_id=self.aws_conn_id)
 
-        job_flow_id = self.job_flow_id
+        emr = emr_hook.get_conn()
 
+        job_flow_id = self.job_flow_id or emr_hook.get_cluster_id_by_name(self.job_flow_name,
+                                                                          self.cluster_states)
         if not job_flow_id:
-            job_flow_id = emr.get_cluster_id_by_name(self.job_flow_name, self.cluster_states)
+            raise AirflowException(f'No cluster found for name: {self.job_flow_name}')
```

Comment:
No I start to think when this exception will be reached? When `emr_hook.get_cluster_id_by_name` return `None`. Is it possible or this will fail when calling `emr_hook.get_cluster_id_by_name` whith `self.job_flow_name=None`?

Reply:
If you ask for a cluster that doesn't exist it returns `None`
https://github.com/apache/airflow/blob/master/airflow/contrib/hooks/emr_hook.py#L52-L59